cmake/CMakeLists.txt

```diff
@@ -43,6 +43,7 @@ option(onnxruntime_GENERATE_TEST_REPORTS "Enable test report generation" OFF)
 option(onnxruntime_ENABLE_STATIC_ANALYSIS "Enable static analysis" OFF)
 option(onnxruntime_ENABLE_PYTHON "Enable python buildings" OFF)
 option(onnxruntime_USE_CUDA "Build with CUDA support" OFF)
+option(onnxruntime_USE_NSYNC "Build with NSYNC support. Only available on Windows" OFF)
```

Comment:
"Only available on windows" - so we're only using this for windows?
but it seems to be a cross platform project?

Reply:
The description was wrong.  It's only on Linux.
nsync itself is cross-platform, but right now, in this pr, I only touched the Linux part.  As Windows has many many more [synchronization primitives](https://docs.microsoft.com/en-us/windows/desktop/sync/synchronization-functions) than Linux, nsync may not help much.  At the first step, I'll focus on the Linux build.